src/node_crypto.cc

```diff
@@ -5568,7 +5568,9 @@ void RandomBytesCheck(RandomBytesRequest* req, Local<Value> (*argv)[2]) {
       memcpy(buf, data, req->size());
       (*argv)[1] = buffer;
     } else {
-      (*argv)[1] = Buffer::New(req->env(), data, size).ToLocalChecked();
+      auto cb = [](char* data, void* hint) {};
+      (*argv)[1] = Buffer::New(req->env(), data, size, cb, nullptr)
```

Comment:
Huh? Why?

Reply:
Because it segfaults with a double free without it!

Reply:
so this particular line of code was not being exercised previously because the input was always a Uint8Array. The if check above would always be true. If any of thing is passed, however, the Buffer would segfault when it tried to GC due to a double free.

Reply:
I will say that I'm not entirely sure this is the right approach to take on this! :-)

Reply:
(The challenge here, btw, is that we have two TypedArray instances being created that cover the same data space... thinking about it now, copying the data would be best.)

Reply:
@jasnell What happens if the original `Uint8Array` is freed before this one?

Reply:
Yep, that's the exact weirdness I'm looking at. Looking at it again, I've just changed the if block above to s/isUint8Array/isArrayBufferView and everything works as it should :-)